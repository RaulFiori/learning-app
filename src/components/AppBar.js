import React from 'react';
import PropTypes from 'prop-types';
import { AppBar as MuiBar, withStyles } from '@material-ui/core';
import Title from './Title';

const styles = theme => {
  const { spacing } = theme;

  return {
    appBar: {
      padding: spacing(2)
    }
  };
};

const AppBar = props => {
  const {
    title,
    classes: { appBar }
  } = props;

  return (
    <MuiBar className={appBar} position="fixed">
      <Title>{title}</Title>
    </MuiBar>
  );
};

AppBar.propTypes = {
  title: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(AppBar);
