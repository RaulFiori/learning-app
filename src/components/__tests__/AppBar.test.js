import React from 'react';

import renderer from 'react-test-renderer';

import AppBar from '../AppBar';

describe('testing app bar', () => {
  test('should throw error with no title', () => {
    const instance = renderer.create(<AppBar />).toJSON();
    expect(instance).toMatchSnapshot();
  });

  test('should render correctly with title', () => {
    const instance = renderer.create(<AppBar title="olÃ¡" />).toJSON();
    expect(instance).toMatchSnapshot();
  });
});
