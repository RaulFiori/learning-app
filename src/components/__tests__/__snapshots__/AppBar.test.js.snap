// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing app bar should render correctly with title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed AppBar-appBar-1"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  >
    ol√°
  </h6>
</header>
`;

exports[`testing app bar should throw error with no title 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed AppBar-appBar-1"
>
  <h6
    className="MuiTypography-root MuiTypography-h6"
  />
</header>
`;
